save-backend/src/main/kotlin/com/saveourtool/save/backend/service/TestSuitesService.kt

```diff
@@ -41,14 +42,35 @@ class TestSuitesService(
     /**
      * Save new test suites to DB
      *
-     * @param testSuitesDto test suites, that should be checked and possibly saved
+     * @param testSuitesDto test suites **from the same source**, that should be checked and possibly saved
      * @return list of *all* TestSuites
      */
+    @Transactional
     @Suppress("TOO_MANY_LINES_IN_LAMBDA", "UnsafeCallOnNullableType")
     fun saveTestSuite(testSuitesDto: List<TestSuiteDto>): List<TestSuite> {
         // FIXME: need to check logic about [dateAdded]
         // It's kind of upsert (insert or update) with key of all fields excluding [dateAdded]
         // This logic will be removed after https://github.com/saveourtool/save-cli/issues/429
+
+        // test suites must be from the same source
+        require(testSuitesDto.map { it.source.name to it.source.organizationName }.distinct().size == 1) {
+            "Do not save test suites from different sources at the same time."
+        }
+
+        // test suites must be from the same commit
+        require(testSuitesDto.map { it.version }.distinct().size == 1) {
+            "Do not save test suites from different commits at the same time."
+        }
+
+        val testSuiteSourceVersion = testSuitesDto.first().version
+        val testSuiteSource = testSuitesDto.first()
+            .let { dto ->
+                testSuitesSourceService.getByName(dto.source.organizationName, dto.source.name)
+            }
+            .apply {
+                latestVersion = testSuiteSourceVersion
```

Comment:
What's this for? Load data and override its version?

Reply:
Yes, we still need to load `testSuiteSource` in order to save `testSuite`s. And we need to update it. Decided to update as soon as I get it from db

Reply:
Oh, so we are setting version from test suites, ok, makes sense. But how do you know we are saving test suiites with the latest version and not fetching from an old commit? Maybe we should keep creation timestamp in test suite source too?

Reply:
For now we are fetching only HEAD of a given branch, so I am not sure that we need to add a new column for that

Reply:
Preprocessor has ability to fetch tests  with a specific version, so potentially we can fetch with old version. For example: tag name vs commit version detected automatically

Reply:
If someone fetched some older version lately, does he expect it to be called *latest*?

Reply:
Renamed to `latestFetchedVersion`